### ISSUE TITLE
Chronos: multiprocess training with nano distribute_backend="subprocess" has a huge accuracy drop

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#4841]
I use training data for evaluation. Only `distribute_backend="subprocess"` reflect the bug.

```python
from bigdl.chronos.data.repo_dataset import get_public_dataset
from bigdl.chronos.forecaster import TCNForecaster
from sklearn.preprocessing import StandardScaler
from bigdl.chronos.metric.forecast_metrics import Evaluator
import time
import numpy as np


def get_tsdata(lookback=96, horizon=24):
    name = 'nyc_taxi'
    tsdata_train, _, tsdata_test = get_public_dataset(name, val_ratio=0, test_ratio=0.4)
    stand_scaler = StandardScaler()
    for tsdata in [tsdata_train, tsdata_test]:
        tsdata.gen_dt_feature(features=["HOUR"], one_hot_features=["HOUR"])\
              .impute(mode="linear")\
              .scale(stand_scaler, fit=(tsdata is tsdata_train))\
              .roll(lookback=lookback, horizon=horizon)
    return tsdata_train, tsdata_test


if __name__ == "__main__":
    tsdata_train, tsdata_test = get_tsdata()
    x_train, y_train = tsdata_train.to_numpy()
    x_test, y_test = tsdata_test.to_numpy()
    print(x_train.shape, y_train.shape, x_test.shape, y_test.shape)
    forecaster = TCNForecaster(past_seq_len = 96,
                               future_seq_len = 24,
                               input_feature_num = x_train.shape[-1],
                               output_feature_num = 1,
                               num_channels = [48] * 7,
                               repo_initialization = False,
                               kernel_size = 3,
                               dropout = 0.1,
                               lr = 0.001,
                               seed = 1)
    forecaster.num_processes = 2
    forecaster.fit((x_train, y_train), epochs=3, batch_size=32)

    st = time.time()
    y_pred = forecaster.predict(x_train)
    fp32_pytorch_time = time.time()-st

    fp32_pytorch_latency = []
    for i in range(x_test.shape[0]):
        x = x_test[i:i+1]
        st = time.time()
        y_pred = forecaster.predict(x)
        fp32_pytorch_latency.append(time.time()-st)


    print("fp32 pytorch smape:", round(float(avg_smape_fp32_pytorch), 2))
```

```bash
# single process
Epoch 2: 100%|████████████████████████████████████████████████████████████████████████████████████████████| 190/190 [00:05<00:00, 33.46it/s, loss=0.0725]
fp32 pytorch mse: 0.06

# 2 process with distribute_backend="subprocess" 
Epoch 2: 100%|████████████████████████████████████████████████████████████████████████████████████████████| 190/190 [00:05<00:00, 34.05it/s, loss=0.0803]
/home/junweid/miniconda3/envs/nano/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/ddp_spawn.py:288: UserWarning: cleaning up ddp environment...
 rank_zero_warn("cleaning up ddp environment...")
fp32 pytorch mse: 1.31

# 2 process with distribute_backend="spawn" 
Epoch 2: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 190/190 [00:05<00:00, 34.40it/s, loss=0.071]
/home/junweid/miniconda3/envs/nano/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/ddp_spawn.py:288: UserWarning: cleaning up ddp environment...
 rank_zero_warn("cleaning up ddp environment...")
fp32 pytorch mse: 0.05
```

**Comment by TheaperDeng** [comment]
@yangw1234 this might be a nano issue. Here is how chronos carry out the inference:
https://github.com/intel-analytics/BigDL/blob/main/python/chronos/src/bigdl/chronos/pytorch/utils.py#L21

**Comment by yangw1234** [comment]
@Django-Jiang could you help look at this issue?

**Comment by Django-Jiang** [comment]
I found training with subprocess will only return 1.31 regardless of epoch number, there should be some problem. I am firstly checking the subprocess part.

**Comment by liangs6212** [comment]
This problem only occurs when `checkpoint_callback` is set to False.
`checkpoint_callback` may be set to False if the user does not need to automatically check ckpt.
For more information, please refer to https://pytorch-lightning.readthedocs.io/en/1.4.2/common/trainer.html#checkpoint-callback
btw, 'checkpoint_callback' will be deprecated.
related doc: https://pytorch-lightning.readthedocs.io/en/1.6.4/common/trainer.html#checkpoint-callback
